feat(total-lines-changed): add total lines added and removed for PR

bertyhell opened the pull request:
This PR adds 2 counters in the files tab header in the right sidebar.
These counters show the total lines added and the total lines removed in the pull request.

![image](https://user-images.githubusercontent.com/1710840/135920358-b9d0b758-c325-466c-984e-ad9e1a61399a.png)

Also correctly adds the counters if there are merge conflicts in the PR:
![image](https://user-images.githubusercontent.com/1710840/135920401-4f2d206e-3046-49cd-851a-ae83d6a6dc62.png)

-   [x] I tested the changes in this pull request myself
-   [x] I added Automated Tests when applicable
-   [x] I added an Option to enable / disable this feature
-   [x] I updated the README.md, with pictures if necessary

bertyhell committed:
feat(total-lines-changed): add total lines added and removed for PR

bertyhell committed:
fix(total-lines-added): add feature to readme

bertyhell committed:
feat(total-lines-changed): make file counter same styles as added/rem.

bertyhell committed:
fix(total-lines-changed): take new screenshot after improving css styles

reyronald commented on src/total-lines-changed/total-lines-changed.js:
```diff
@@ -0,0 +1,76 @@
+
+const FILES_TAB_SELECTOR = '[data-testid="sidebar-tab-files"]'
+
+let totalAdded: number
+let totalRemoved: number
```
Is it necessary for this to be out here at module scope level? Can't it be within the scope of the function ?

If you navigate from one PR to a different one, is the count accurate in both ? I don't see these values being reset anywhere

reyronald reviewed (commented):
Great work! Thanks a lot for contributing this, also you did an impressive job sticking to the existing patterns in the codebase.

Just three questions

1. How does the UI look when the number of changes is a really high number ? E.g. > 100, > 1000, > 10000
2. Is it possible for the response of the API to not contain the entire amount file changes in the PR ? If it is, can we tell with the response, maybe by looking at the `size` property? Should we add some indicator in the UI that there might be more lines changed that we are displaying in that case ? E.g. `+ >9,999`

bertyhell committed:
fix(total-lines-changed): add K, M lines changed, stop if # files > 1000

bertyhell commented:
> 1. How does the UI look when the number of changes is a really high number ? E.g. > 100, > 1000, > 10000

I've added improvements for this:
![image](https://user-images.githubusercontent.com/1710840/135972379-6aacfff0-7832-42aa-b1d6-32c14a3b284b.png)



> 2\. Is it possible for the response of the API to not contain the entire amount file changes in the PR ?

Yes, we currently only fetch the first 1000 files. So if the size is larger than the total lines will be incorrect. I've added some code to detect this and omit the calculation when this happens.


> Is it necessary for this to be out here at module scope level? Can't it be within the scope of the function ?

No this was a remnant of some older code.


> If you navigate from one PR to a different one, is the count accurate in both ? I don't see these values being reset anywhere

Yes, I tried adding a link to another PR inside the first PR. When i click it, the whole page is reloaded, so the whole code is executed again. This produces the correct result

bertyhell committed:
fix(total-lines-changed): fix flow errors

bertyhell committed:
Merge remote-tracking branch 'refined-bitbucket/dev' into feature/total-lines-changed

bertyhell committed:
fix(total-lines-changed): fix build issues with nesting depth

bertyhell committed:
fix(total-lines-changed): fix xo errors

bertyhell committed:
fix(total-lines-changed): fix pullrequest commit amount test

reyronald reviewed (approved):
This looks great, thanks a lot!
